packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -71,28 +99,57 @@ library LibProposing {
         meta.depositsProcessed =
             LibEthDepositing.processDeposits(state, config, input.beneficiary);
 
-        unchecked {
-            meta.timestamp = uint64(block.timestamp);
-            meta.l1Height = uint64(block.number - 1);
-            meta.l1Hash = blockhash(block.number - 1);
-            meta.mixHash = bytes32(block.difficulty * state.numBlocks);
-        }
-
+        // Init the block
         TaikoData.Block storage blk =
-            state.blocks[state.numBlocks % config.ringBufferSize];
+            state.blocks[state.numBlocks % config.blockRingBufferSize];
 
+        blk.metaHash = LibUtils.hashMetadata(meta);
         blk.blockId = state.numBlocks;
-        blk.proposedAt = meta.timestamp;
+        blk.gasLimit = meta.gasLimit;
         blk.nextForkChoiceId = 1;
         blk.verifiedForkChoiceId = 0;
-        blk.metaHash = LibUtils.hashMetadata(meta);
+        blk.proverReleased = false;
+
         blk.proposer = msg.sender;
+        blk.feePerGas = state.feePerGas;
+        blk.proposedAt = meta.timestamp;
 
-        uint64 blockFee;
+        if (assignedProver == address(0)) {
+            if (state.numOpenBlocks >= config.rewardOpenMaxCount) {
+                revert L1_TOO_MANY_OPEN_BLOCKS();
+            }
+            blk.rewardPerGas = state.feePerGas;
+            ++state.numOpenBlocks;
```

Comment:
Where is the code to open this block?

Reply:
If the `assignProver` returns `address(0)`, then the block is considered as an open block.

Reply:
Inside LibProposing:

```solidity
 if (assignedProver == address(0)) {
            if (state.numOpenBlocks >= config.rewardOpenMaxCount) {
                revert L1_TOO_MANY_OPEN_BLOCKS();
            }
            blk.rewardPerGas = state.feePerGas;
            ++state.numOpenBlocks;
        } 
```